docs: Adding the missed documentation for the RBAC (#4515)
feast-dev/feast: docs/getting-started/components/authz_manager.md

# Authorization Manager

> An Authorization Manager is an instance of the `AuthManager` class that is plugged into one of the Feast servers to extract user details from the current request and inject them into the permission framework.

Reviewer: why remove link?

Author: original link is broken (extra s in the file permissions.md). Thought we moved that doc is moved but with your comment realized it is type or some refactoring. now fixed it with the correct link.

Some assumptions are made in the OIDC server configuration:
* The OIDC token refers to a client with roles matching the RBAC roles of the configured `Permission`s (*)

> * The roles are exposed in the access token passed to the server 

Reviewer: ```suggestion
* The roles are exposed in the access token that is passed to the server 
```

> * JWT token is expected to have verified signature and not expired. Feast oidc token parser logic validates for `verify_signature` and `verify_exp` so make sure that given OIDC provider is configured to meet this requirement.

Reviewer: ```suggestion
* The JWT token is expected to have a verified signature and not be expired. The Feast OIDC token parser logic validates for `verify_signature` and `verify_exp` so make sure that the given OIDC provider is configured to meet these requirements.
```

> * preferred_username should be part of the jwt token claim.

Reviewer: ```suggestion
* The preferred_username should be part of the JWT token claim.
```
